src/utils/httpClient.ts

```diff
@@ -0,0 +1,36 @@
+import axios from 'axios'
+
+export type AxiosDNSLookupFunction = typeof axios.defaults.lookup
+
+export const httpClient = axios.create({})
+
+export const sendAxiosRequestWithRequestInit = async (url: string | URL, requestInit: RequestInit) => {
+  const { method, headers, ...rest } = requestInit
+
+  if (url instanceof URL) {
+    url = url.toString()
+  }
+
+  return axios.request({
+    url,
+    method,
+    headers: createAxiosHeadersFromHeadersInit(headers),
+    ...rest,
+  })
+}
+
+export const createAxiosHeadersFromHeadersInit = (headersInit: HeadersInit): Record<string, string> => {
+  if (headersInit instanceof Headers) {
+    return Object.fromEntries(headersInit.entries())
+  }
+
+  if (Array.isArray(headersInit)) {
+    return Object.fromEntries(headersInit)
+  }
+
+  if (typeof headersInit === 'object' && headersInit !== null) {
```

Comment:
I think this additional check might not be necessary because, aside from `Headers` and `array` types, it seems only `Record` can be passed here. So, I think we can simply use `return { ...headersInit }`.